feat: dedicated config processing (#6400)
Kong/docs.konghq.com: app/_src/gateway/production/sizing-guidelines.md

scenario, adding more cache memory would not increase maximum throughput by
much.

{% if_version gte:3.5.x %}
In {{site.base_gateway}} setups with high cardinality of configured Routes, Services,

> Plugins, Consumers etc, the cluster may benefit from "Dedicated Config Processing". The

Reviewer: ```suggestion
Plugins, Consumers etc, the cluster may benefit from dedicated config processing. This
```

feature presents an innovative approach to enhance system performance by employing a dedicated
worker process. By offloading specific tasks, it promises reduced latencies and offers a more

> efficient operational experience, when operating in `hybrid mode``.

Reviewer: ```suggestion
efficient operational experience, when operating in `hybrid mode`.
```

> In {{site.base_gateway}} setups with high cardinality of configured Routes, Services,
> Plugins, Consumers etc, the cluster may benefit from "Dedicated Config Processing". The
> feature presents an innovative approach to enhance system performance by employing a dedicated
> worker process. By offloading specific tasks, it promises reduced latencies and offers a more
> efficient operational experience, when operating in `hybrid mode``.
> Users should remain aware of the slight increase in memory usage but can benefit significantly
> from the accelerated processing times.

Reviewer 2: ```suggestion
When {{site.base_gateway}} is operating in `hybrid` mode with a large number of configuration 
entities (routes, services etc.), it can benefit from the "Dedicated Configuration Processing" option.  
When enabled, certain CPU  intensive steps of the data plane reconfiguration operation are offloaded 
to a dedicated worker process.  This reduces proxy latency during reconfigurations at the cost of a 
slight increase in memory usage.  The benefits of this mechanism are most apparent with configurations
consisting of more than a thousand configuration entities.
```

> (add_me: link to "Reference -> kong.conf -> "dedicated config processing")

Author: @Kong/docs-maintainers  Two questions:

1. Do we update the `References -> kong.conf` section automatically with new config options?

2. How do you link to this section?

Reviewer: > Do we update the References -> kong.conf section automatically with new config options?

Pseudo-automatically, but yes. We'll have that generated today.

> How do you link to this section?

The link would be `/gateway/{{page.kong_version}}/reference/configuration/#dedicated_config_processing`.

Author: ty, added
